### PR TITLE
Add batch examples to READMEs

### LINKED ISSUE #46: Initialization of Contracts is Vulnerable to Front-running
Impact: High
Likelihood: Medium
Reported by a third-party.

**Description**

There are two actions required to deploy and initialize a contract. The first is the deployment of the code which involves uploading the contract byte code ( .wasm file). The second is the initialization function (e.g. new() or default() ), which sets the account state storage based on the code in the contract. The initialization action may be called from any account.

If these actions are not part of a single transaction, it is possible to front-run the initialization transaction.

When an attacker witnesses a transaction to deploy the contract they may wait for the execution of the transaction, then front-run the initialization transaction (i.e. the call to new() ) with their own parameters. Front-running of transactions in this scenario is trivial for any block producer as they may decide the order of the transactions inserted into a new block and may always create the following scenario:
1. Deployment transaction;
2. Malicious initialisation transaction;
3. Owner initialisation transaction.

In the case of contract, an attacker can front-run the call to new() . This will allow the attacker to set the , the   and the transfers to be enabled. The attacker may then transfer the entire balance of the contract to their personal account.

In the case of contract, an attacker can front-run the call to new() . This will allow the attacker to set the   parameter, gaining ownership of the contract. The contract is required to be initialised with an initial balance which cannot be attached to the call to new() as it is not payable. The attacker will therefore receive all staking rewards fees associated with the contract, as they may set the reward percentage to 100%. Therefore, the fees received from this initial balance will be rewarded to the new owner.

In the case of the contract, the front runner can here again front-run the new() transaction call. The attacker may also set the parameter which will allocate them the entire token supply.

**Recommendations**
Ensure there is adequate documentation in all cases to warn users against the dangers of using multiple trans- actions. When deploying new contracts, developers should always initialize contracts in the same transaction.

### PR DESCRIPTION
**Pull request by willemneal**
Partially addresses #46

### CODE REVIEW

File: lockup/README.md [hunk 1]
```diff
@@ -457,3 +457,57 @@ near call owner1 termination_withdraw '{"receiver_id": "near"}' --accountId=near
...
+const account = await near.account("ownerID");
+const contractName = "lockup.ownerID";
+const newArgs = {
+        lockup_duration: new BigInt(1000000000),
+        lockup_start_information: {
+            transfer_poll_account_id: "transfer-poll",
+        },
+        initial_owners_main_public_key: "Eg2jtsiMrprn7zgKKUk79qM1hWhANsFyE6JSX4txLEuy",
+        foundation_account_id: null,
+        staking_pool_whitelist_account_id: "whitelistID",
```
Comment (evgenykuzyakov) [inline lockup/README.md:503]: whitelistID is invalid name
```suggestion
        staking_pool_whitelist_account_id: "staking-pools-whitelist",
```

File: lockup/README.md [hunk 2]
```diff
@@ -457,3 +457,57 @@ near call owner1 termination_withdraw '{"receiver_id": "near"}' --accountId=near
...
+}
+const result = account.signAndSendTransaction(
+    contractName,
+    [
+        nearAPI.transactions.createAccount(),
+        nearAPI.transactions.transfer("100000000000000000000000000"), 
+        nearAPI.transactions.deployContract(fs.readFileSync("res/multisig.wasm")),
```
Comment (evgenykuzyakov) [inline lockup/README.md:510]:
```suggestion
        nearAPI.transactions.deployContract(fs.readFileSync("res/lockup_contract.wasm")),
```

File: lockup/README.md [hunk 3]
```diff
@@ -457,3 +457,57 @@ near call owner1 termination_withdraw '{"receiver_id": "near"}' --accountId=near
...
+        outcome_into_result(res)
+    }
+```
+
+To do this with near shell, the `near repl` command can be used and the following executed in the REPL.
+
+```js
+const fs = require('fs');
+const account = await near.account("ownerID");
+const contractName = "lockup.ownerID";
```
Comment (evgenykuzyakov) [inline lockup/README.md:495]:
```suggestion
const account = await near.account("foundation");
const contractName = "lockup-owner-id";
```

File: lockup/README.md [hunk 4]
```diff
@@ -457,3 +457,57 @@ near call owner1 termination_withdraw '{"receiver_id": "near"}' --accountId=near
...
+const account = await near.account("foundation");
+const contractName = "lockup-owner-id";
+const newArgs = {
+        lockup_duration: new BigInt(1000000000),
+        lockup_start_information: {
+            transfer_poll_account_id: "transfer-poll",
+        },
+        initial_owners_main_public_key: "Eg2jtsiMrprn7zgKKUk79qM1hWhANsFyE6JSX4txLEuy",
+        foundation_account_id: null,
+        staking_pool_whitelist_account_id: "staking-pools-whitelist",
```
Comment (evgenykuzyakov) [inline lockup/README.md:503]: That's valid input from README example
```suggestion
    "lockup_duration": "31536000000000000",
    "lockup_start_information": {
        "TransfersDisabled": {
            "transfer_poll_account_id": "transfers-poll"
        }
    },
    "vesting_schedule": {
        "start_timestamp": "1535760000000000000",
        "cliff_timestamp": "1567296000000000000",
        "end_timestamp": "1661990400000000000"
    },
    "staking_pool_whitelist_account_id": "staking-pool-whitelist",
    "initial_owners_main_public_key": "KuTCtARNzxZQ3YvXDeLjx83FDqxv2SdQTSbiq876zR7",
    "foundation_account_id": "near"
```

File: lockup/README.md [hunk 5]
```diff
@@ -457,3 +457,57 @@ near call owner1 termination_withdraw '{"receiver_id": "near"}' --accountId=near
...
+}
+const result = account.signAndSendTransaction(
+    contractName,
+    [
+        nearAPI.transactions.createAccount(),
+        nearAPI.transactions.transfer("100000000000000000000000000"), 
+        nearAPI.transactions.deployContract(fs.readFileSync("res/lockup_contract.wasm")),
+        nearAPI.transactions.functionCall("new", Buffer.from(JSON.stringify(newArgs)), 10000000000000, "0"),
```
Comment (evgenykuzyakov) [inline lockup/README.md:511]:
```suggestion
        nearAPI.transactions.functionCall("new", Buffer.from(JSON.stringify(newArgs)), 100000000000000, "0"),
```

**Comment by willemneal** [comment]
I thought it might make more sense to add it to the top level README.

Although perhaps it would be better to add valid deploy scripts for each contract.

**Pull request merged**